lib/dns_cluster.ex

```diff
@@ -58,6 +58,27 @@ defmodule DNSCluster do
         {:error, _} -> []
       end
     end
+
+    def lookup(query, type) when is_binary(query) and type in [:srv] do
+      case :inet_res.getbyname(~c"#{query}", type) do
+        {:ok, hostent(h_addr_list: srv_list)} ->
+          lookup_hosts(srv_list)
+
+        {:error, _} ->
+          []
+      end
+    end
+
+    defp lookup_hosts(srv_list) do
+      srv_list
+      |> Enum.flat_map(fn {_prio, _weight, _port, host_name} ->
+        case :inet.gethostbyname(host_name) do
+          {:ok, hostent(h_addr_list: addr_list)} -> addr_list
+          {:error, _} -> []
+        end
+      end)
+      |> Enum.filter(& &1)
```

Comment:
Can an entry in `h_addr_list` ever be nil or false? :) Or can we remove this filter?

Reply:
Ah true, thanks I had the error case return nil before. That filter does not make sense anymore.